internal/datastore/proxy/mapping.go

```diff
@@ -179,18 +180,52 @@ func (mp mappingProxy) DeleteNamespace(ctx context.Context, nsName string) (data
 	return mp.delegate.DeleteNamespace(ctx, storedNamespaceName)
 }
 
-func (mp mappingProxy) QueryTuples(filter datastore.TupleQueryResourceFilter, revision datastore.Revision) datastore.TupleQuery {
-	var err error
+func (mp mappingProxy) QueryTuples(
+	ctx context.Context,
+	filter *v1.RelationshipFilter,
+	revision datastore.Revision,
+	opts ...options.QueryOptionsOption,
+) (datastore.TupleIterator, error) {
 	resourceType, err := mp.mapper.Encode(filter.ResourceType)
-	return mappingTupleQuery{
-		mp.delegate.QueryTuples(datastore.TupleQueryResourceFilter{
-			ResourceType:             resourceType,
-			OptionalResourceID:       filter.OptionalResourceID,
-			OptionalResourceRelation: filter.OptionalResourceRelation,
-		}, revision),
-		mp.mapper,
-		err,
+	if err != nil {
+		return nil, fmt.Errorf(errTranslation, err)
+	}
+
+	var subFilter *v1.SubjectFilter
+	if filter.OptionalSubjectFilter != nil {
+		subResourceType, err := mp.mapper.Encode(filter.OptionalSubjectFilter.SubjectType)
+		if err != nil {
+			return nil, fmt.Errorf(errTranslation, err)
+		}
+
+		subFilter = &v1.SubjectFilter{
+			SubjectType:       subResourceType,
+			OptionalSubjectId: filter.OptionalSubjectFilter.OptionalSubjectId,
+			OptionalRelation:  filter.OptionalSubjectFilter.OptionalRelation,
+		}
+	}
+
+	queryOpts := options.NewQueryOptionsWithOptions(opts...)
```

Comment:
Can we add a check in here that the *only* two options used are Limit and/or Usersets? That way, if we add another new option down the road, we don't forget to map it

Reply:
ideas on how we would do that?

Reply:
Have optgen generate some sort of count of the number of options possible and then panic here if it mismatches? Alternatively, optgen could generate a unique "type" or constant name encompassing the options available, and then we reference it here, ensuring that if a new options is added, that reference no longer type checks?